### PR TITLE
[airq] Fix occasional stalling of sensor value updates and refactoring

### LINKED ISSUE #17164: [airq] Communication stalls occasionally
Under rare conditions the response from the device cannot be parsed, leading to a crash of the binding. The Thing keeps being online, but the Channels are not updated anymore.

This is the stack trace:
```
2024-07-27 14:58:37.671 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.StringIndexOutOfBoundsException: begin 1, end -1, length 4
        at java.lang.String.checkBoundsBeginEnd(String.java:4606) ~[?:?]
        at java.lang.String.substring(String.java:2709) ~[?:?]
        at org.openhab.binding.airq.internal.AirqHandler$ResultPair.<init>(AirqHandler.java:108) ~[?:?]
        at org.openhab.binding.airq.internal.AirqHandler.processType(AirqHandler.java:610) ~[?:?]
        at org.openhab.binding.airq.internal.AirqHandler.pollData(AirqHandler.java:470) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
        at java.lang.Thread.run(Thread.java:840) [?:?]
```
I'm looking into it.

### PR DESCRIPTION
**Pull request by fwolter**
In rare cases the sensor sends a null value in the JSON data, which lead to a crash of the poller task.

Also:
- Remove trace debugging statements
- Handle InterruptedException correctly
- Replace substring parsing by GSON functionality

Fixes #17164

### CODE REVIEW

File: bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java [hunk 1]
```diff
@@ -439,65 +439,70 @@ public void dispose() {
...
+                processType(decObj, "performance", "performance", "number");
+                processType(decObj, "timestamp", "timestamp", "datetime");
+                processType(decObj, "uptime", "uptime", "numberTimePeriod");
+                processType(decObj, "tvoc", "tvoc", "pairPPB");
+
+                updateStatus(ThingStatus.ONLINE);
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Syntax error while parsing response from device: " + e.getMessage());
+                logger.error("Parse error in response: {}", jsonAnswer);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java:504]: High level logging is already performed by setting the thingstate. When verbose message contents are logged, the trace level should be used.

```suggestion
                logger.trace("Parse error in response: {}", jsonAnswer);
```

**Comment by fwolter** [comment]
Thanks for the review! I addressed the feedback.

**Review by lsiepel — APPROVED** [review]
Thanks,LGTM

**Comment by fwolter** [comment]
Thanks!

**Pull request merged**